### DISCUSSION TITLE
Can pyhanko add picture to visible signature field

### DISCUSSION THREAD

**Discussion by deaw0003** [MatthiasValvekens/pyHanko#285] [Q&A] [answered]
Hi. I want to know. Can pyhanko add picture to visible  signature field. My usecase is I want to add picture of real handwriting to visible signature field. This is my example code that I use to add visible signature field but I don't know how to add a picture to it.

```py
pdf_signer = signers.PdfSigner(
    signers.PdfSignatureMetadata(
        field_name="sig_test",
        subfilter=fields.SigSeedSubFilter.PADES,
        md_algorithm="sha256",
    ),
    signer=ext_signer,
    new_field_spec=fields.SigFieldSpec(
        "sig_test", on_page=-1, box=[10, 74, 140, 134]
    )
)
```

**Comment by MatthiasValvekens** [comment, marked as answer]
Yes. One way to do this is by using the `background` property of a `TextStampStyle`: https://pyhanko.readthedocs.io/en/latest/lib-guide/signing.html#text-based-stamps.

For more advanced use cases, you can also use `StaticStampStyle` and feed it whatever you want: https://pyhanko.readthedocs.io/en/latest/lib-guide/signing.html#static-content-stamps.

**Comment by deaw0003** [comment]
Thank you very much. You really help me a alot.
